packages/marketplace-ui/src/components/TokenItem.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  Text,
+  Flex,
+  Image,
+  Icon,
+} from "@chakra-ui/react";
+import { MdSettings } from "react-icons/md";
+import { PublicKey } from "@solana/web3.js";
+import React, { useMemo } from 'react';
+import { useWallet } from "@solana/wallet-adapter-react";
+import { useTokenAuthorities, useTokenBondingFromMint } from "@strata-foundation/react";
+import { useRouter } from "next/router";
+import { routes, route } from "../utils/routes";
+import { MetadataData } from "@metaplex-foundation/mpl-token-metadata";
+
+export const TokenItem = ({mint, isIntermediateToken}: {
+  mint: PublicKey,
+  isIntermediateToken: (mint: PublicKey, metadata: MetadataData, hasTokenBonding: boolean) => boolean
+}) => {
+  const router = useRouter();
+  const { publicKey } = useWallet();
+  const { data, hasAnyAuth, image, metadata } = useTokenAuthorities(mint, publicKey || undefined);
+  const { info: tokenBonding } = useTokenBondingFromMint(mint);
+
+  const shouldDisplay = mint && metadata && hasAnyAuth ? !isIntermediateToken(mint, metadata!, !!tokenBonding) : false;
+
+  return (
+    <>
+      { hasAnyAuth && shouldDisplay ? (
```

Comment:
nit: hasAnyAuth is already in shouldDisplay. Also a bit confusing using the ternary operator on a pure boolean. Dunno if there's a good way to clean this up?

(for context, nit means nitpick, don't need to fix but if you're fixing other junk may be nice)

Reply:
True. Why is it confusing to use a ternary on a pure bool?

Reply:
Usually you can rewrite the ternary to just be fully boolean logic, since ternary is effectively a bool. To me it reads like

   if (something) {
      return true
   } else {
      return false
   }

Reply:
Fixed